crates/trie/src/state.rs

```diff
@@ -38,17 +39,27 @@ impl HashedPostState {
         state: impl IntoIterator<Item = (&'a Address, &'a BundleAccount)>,
     ) -> Self {
         let mut this = Self::default();
-        for (address, account) in state {
-            let hashed_address = keccak256(address);
-            this.accounts.insert(hashed_address, account.info.clone().map(into_reth_acc));
-
-            let hashed_storage = HashedStorage::from_iter(
-                account.status.was_destroyed(),
-                account.storage.iter().map(|(key, value)| {
-                    (keccak256(B256::new(key.to_be_bytes())), value.present_value)
-                }),
-            );
-            this.storages.insert(hashed_address, hashed_storage);
+
+        let changes = state
+            .into_iter()
+            .collect::<Vec<(&'a Address, &'a BundleAccount)>>()
+            .into_par_iter()
+            .map(|(address, account)| {
+                let hashed_address = keccak256(address);
+                let hashed_account = account.info.clone().map(into_reth_acc);
+                let hashed_storage = HashedStorage::from_iter(
+                    account.status.was_destroyed(),
+                    account.storage.iter().map(|(key, value)| {
+                        (keccak256(B256::new(key.to_be_bytes())), value.present_value)
```

Comment:
this could also be parallelized

Reply:
i dont think it's that straight forward, i wouldnt be surprised that the length of storage keys that need to be hashed to be bigger than the number of accounts. and if the number of accounts requires parallalization, this would too

Reply:
but as with the PR itself, it would require some benchmark comparisons